packages/@lwc/engine/src/framework/restrictions.ts

```diff
@@ -393,51 +319,15 @@ function getCustomElementRestrictionsDescriptors(
                 return originalAddEventListener.apply(this, arguments);
             },
         }),
-        // replacing mutators and accessors on the element itself to catch any mutation
-        setAttribute: generateDataDescriptor({
-            value: setAttributePatched,
-        }),
-        setAttributeNS: generateDataDescriptor({
-            value: setAttributeNSPatched,
-        }),
-        removeAttribute: generateDataDescriptor({
-            value: removeAttributePatched,
-        }),
-        removeAttributeNS: generateDataDescriptor({
-            value: removeAttributeNSPatched,
-        }),
     });
 }
 
-function getComponentRestrictionsDescriptors(cmp: ComponentInterface): PropertyDescriptorMap {
+function getComponentRestrictionsDescriptors(): PropertyDescriptorMap {
     if (process.env.NODE_ENV === 'production') {
         // this method should never leak to prod
         throw new ReferenceError();
     }
-    const originalSetAttribute = cmp.setAttribute;
     return {
-        setAttribute: generateDataDescriptor({
-            value(this: ComponentInterface, attrName: string, _value: any) {
-                if (isString(attrName)) {
-                    const propName = getPropNameFromAttrName(attrName);
-                    const globalAttrName =
-                        globalHTMLProperties[propName] && globalHTMLProperties[propName].attribute;
-                    // Check that the attribute name of the global property is the same as the
-                    // attribute name being set by setAttribute.
-                    if (attrName === globalAttrName) {
-                        const { error } = globalHTMLProperties[propName];
```

Comment:
These errors are meant for when global HTML properties are accessed (e.g., `this.dataset`).